pkg/visor/init.go

```diff
@@ -1179,9 +1182,35 @@ func getPublicIP(v *Visor, service string) (pIP string, err error) {
 		return pIP, fmt.Errorf("provided URL is invalid: %w", err)
 	}
 
-	// Wait until stun client is ready
-	<-v.stunReady
+	pIP, err = getip2()
+	if err != nil {
+		return pIP, fmt.Errorf("cannot fetch public ip")
+	}
 
-	pIP = v.stunClient.PublicIP.IP()
 	return pIP, nil
 }
+
+type ipAPI struct {
+	Query string
+}
+
+func getip2() (string, error) {
+	req, err := http.Get("http://ip-api.com/json/")
```

Comment:
we should use our own service here `ip.skycoin.com`

Reply:
Fixed.